internal/controller/instaslice_daemonset.go

```diff
@@ -435,6 +435,10 @@ func (r *InstaSliceDaemonsetReconciler) discoverMigEnabledGpuWithSlices() ([]str
 		os.Exit(1)
 	}
 	log.Info("classical resources obtained are ", "cpu", cpu, "memory", memory)
+	if instaslice == nil {
```

Comment:
this is nil probably because errorDiscoveringProfiles isn't handled above.

Reply:
Added error handling @rphillips